src/main/telemetry/smartport.c

```diff
@@ -348,19 +349,12 @@ void handleSmartPortTelemetry(void)
                     // the MSB of the sent uint32_t helps FrSky keep track
                     // the even/odd bit of our counter helps us keep track
                     if (smartPortIdCnt & 1) {
-                        tmpui = tmpi = GPS_coord[LON];
-                        if (tmpi < 0) {
-                            tmpui = -tmpi;
-                            tmpui |= 0x40000000;
-                        }
-                        tmpui |= 0x80000000;
+                        tmpui = (abs(GPS_coord[LON] / 100) * 6) | 0x80000000;
```

Comment:
`GPS_coord[LON] * 6 / 100` will probably get more accurate result (dividing first truncates result to multiple of six)